ocp_resources/machine_config_pool.py

```diff
@@ -2,8 +2,74 @@
 
 
 class MachineConfigPool(Resource):
+    """
+    MachineConfigPool object. API reference:
+    https://docs.openshift.com/container-platform/4.12/rest_api/machine_apis/machineconfigpool-machineconfiguration-openshift-io-v1.html
+
+    Args:
+        machine_config_selector (object): Matching labels/expressions, to determine which MachineConfig objects
+            to apply this MachineConfigPool object.
+        configuration (dict): Targeted MachineConfig object for the machine config pool, in the following format:
+            {"name": (str), "source": <List of dicts, each representing a MachineConfig resource>}
+        node_selector_labels (object): Matching labels, to determine to which nodes to apply this
+            MachineConfigPool object.
+        node_selector_expressions (object): Matching expressions, to determine to which nodes to apply this
```

Comment:
should be `machine_config_selector`
same logic as `node_selector`

Reply:
Sorry, I don't understand this comment.
What do you mean by "should be machine_config_selector"?
These are members of nodeSelector.

Reply:
`MachineConfigPool` has both `machineConfigSelector` and `nodeSelector`:
```
  machineConfigSelector:
    matchLabels:
      machineconfiguration.openshift.io/role: master
  nodeSelector:
    matchLabels:
      node-role.kubernetes.io/master: ""
```

Each can have different options for selection logic.
Selection logic has `matchLabels` and other options as well

Reply:
I hope I got what you mean this time.